res_handler_transform.go

```diff
@@ -103,11 +103,23 @@ func (h *ResponseTransformMiddleware) HandleResponse(rw http.ResponseWriter, res
 			logger.WithError(err).Error("Error unmarshalling XML")
 		}
 	default: // apidef.RequestJSON
-		if err := json.NewDecoder(respBody).Decode(&bodyData); err != nil {
+		var tempBody interface{}
+		if err := json.NewDecoder(respBody).Decode(&tempBody); err != nil {
 			logger.WithError(err).Error("Error unmarshalling JSON")
 		}
+
+		switch tempBody.(type) {
+		case []interface{}:
+			logger.Debug("got array: []interface{}")
+			bodyData["array"] = tempBody
+		case map[string]interface{}:
```

Comment:
Do not think you can do this. Type checking for `map[string]interface{}` does not work. You should use `reflect.TypeOf(a).Kind() == reflect.Map` and `reflect.TypeOf(a).Kind() == reflect.Array`

Reply:
I think it should be OK as json un-marshalling does `map[string]interface{}` or `[]interface{}` if not strict type provided.

It can also be done using type assertion:
```
if tmpVal, ok := tempBody.([]interface{}); ok {
		fmt.Println("got array: []interface{}", tmpVal)
	} else if tmpVal, ok := tempBody.(map[string]interface{}); ok {
		fmt.Println("got object: map[string]interface{}", tmpVal)
	} else {
		fmt.Println("Got something else", tmpVal)
	}
```

also, I think any of those three have about the same performance